server/internal/router/router.go

```diff
@@ -0,0 +1,192 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	v1 "github.com/llm-operator/inference-manager/api/v1"
+	"github.com/llm-operator/inference-manager/server/internal/config"
+	"github.com/llm-operator/inference-manager/server/internal/k8s"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const tickPeriod = 60 * time.Second
+
+type model struct {
+	id string
+}
+
+// R manages the routing table of models to engines.
+type R struct {
+	// m is the route map.
+	m             *routeMap
+	engineClients map[string]v1.InferenceEngineInternalServiceClient
+	imeConfig     config.InferenceManagerEngineConfig
+}
+
+// New creates a new router.
+func New(c config.InferenceManagerEngineConfig) *R {
+	return &R{
+		m:             newRouteMap(),
+		engineClients: make(map[string]v1.InferenceEngineInternalServiceClient),
+		imeConfig:     c,
+	}
+}
+
+// Run updates the routes periodically and upon pod events.
+func (r *R) Run(ctx context.Context, errCh chan error) error {
+	coreClient, err := k8s.NewCoreClient()
+	if err != nil {
+		return fmt.Errorf("create K8sClient: %v", err)
+	}
+
+	stopper := make(chan struct{})
+	defer close(stopper)
+	defer runtime.HandleCrash()
+
+	engineLabels := map[string]string{
```

Comment:
This is a personal preference, so please feel free to ignore, but it would be easier to read if we create `engineLabels` inside `refreshRoutes`. As this is also used in the informer, we can also define a helper function to construct `engineLabels`.